lib/gosu/channel.rb

```diff
@@ -24,39 +24,46 @@ def __pointer
     end
 
     def playing?
-      _channel_playing(@__channel)
+      Gosu_Channel_playing(@__channel).tap { Gosu.check_last_error }
     end
 
     def pause
-      _channel_pause(@__channel)
+      Gosu_Channel_pause(@__channel)
+      Gosu.check_last_error
     end
 
     def paused?
-      _channel_paused(@__channel)
+      Gosu_Channel_paused(@__channel).tap { Gosu.check_last_error }
     end
 
     def resume
-      _channel_resume(@__channel)
+      Gosu_Channel_resume(@__channel)
+      Gosu.check_last_error
     end
 
     def stop
-      _channel_stop(@__channel)
+      Gosu_Channel_stop(@__channel)
+      Gosu.check_last_error
     end
 
     def volume=(double)
-      _channel_set_volume(@__channel, double.clamp(0.0, 1.0))
+      Gosu_Channel_set_volume(@__channel, double.clamp(0.0, 1.0))
+      Gosu.check_last_error
     end
 
     def speed=(double)
-      _channel_set_speed(@__channel, double.clamp(0.0, 1.0))
+      Gosu_Channel_set_speed(@__channel, double.clamp(0.0, 1.0))
+      Gosu.check_last_error
     end
 
     def pan=(double)
-      _channel_set_pan(@__channel, double.clamp(0.0, 1.0))
+      Gosu_Channel_set_pan(@__channel, double.clamp(-1.0, 1.0))
+      Gosu.check_last_error
     end
-  end
 
-  def free_object
-    _destroy_channel(@__channel)
+    def free_object
```

Comment:
@cyberarm This free_object method was defined on the Gosu module, not inside Gosu::Channel. I haven't had time to look at FFI memory management yet, not sure why Channel and Image have different approaches

Reply:
Sounds good.